### PR TITLE
Added Local RPC Command Line & Python Client 

### PR DESCRIPTION
**Pull request by creslinux**
Added only the "Daily" call so far, submitting for early review/feedback

This depends on zerorpc as a requirement.
simple examples here:
[http://www.zerorpc.io/](http://www.zerorpc.io/)
Installed with pip3 install zerorpc

localRCP is enabled/disabled from within config.json. E.g
```
    "localrpc": {
        "enabled": true
       },
```

The server is enabled from within existing rpc manager
and makes use of the existing superclass (RPC)

Through making use of the existing hard work done in rpc.py
It *should be easy to add the other Telegram calls into local_rpc_server.py

The server is wrapped in a thread to be non-blocking
The server and client accept serialised calls or not, used in daily to return json
The client can be used from command line or in a python client script

As example, from cmdline for last 3 days Daily

> /Users/creslin/PycharmProjects/freqtrade_new/.env/bin/zerorpc tcp://127.0.0.1:4242 daily 3
> connecting to "tcp://127.0.0.1:4242"
> False
> ('[\n'
>  '    [\n'
>  '        "2018-06-08",\n'
>  '        "0.00000000 BTC",\n'
>  '        "0.000 USDT",\n'
>  '        "0 trade"\n'
>  '    ],\n'
>  '    [\n'
>  '        "2018-06-07",\n'
>  '        "0.00000000 BTC",\n'
>  '        "0.000 USDT",\n'
>  '        "0 trade"\n'
>  '    ],\n'
>  '    [\n'
>  '        "2018-06-06",\n'
>  '        "0.00000000 BTC",\n'
>  '        "0.000 USDT",\n'
>  '        "0 trade"\n'
>  '    ]\n'
>  ']')
> 

Programatically this would be a client such like:
```
import zerorpc
c = zerorpc.Client()
c.connect("tcp://127.0.0.1:4242")
for item in c.daily(3):
    print item
```

Thank you for sending your pull request. But first, have you included
unit tests, and is your code PEP8 conformant? [More details](https://github.com/freqtrade/freqtrade/blob/develop/CONTRIBUTING.md)

## Summary
Explain in one sentence the goal of this PR

Solve the issue: #___

## Quick changelog

- <change log #1>
- <change log #2>

## What's new?
*Explain in details what this PR solve or improve. You can include visuals.*

### CODE REVIEW

**Comment by creslinux** [comment]
zerorpc homepage 
https://github.com/0rpc/zerorpc-python

**Comment by xmatthias** [comment]
I personally think rest would be a more versitile service/protocol to add - as it would allow to add 
* a web frontent
* a client-script controlling the bot

while rpc is (probably) limited to a client-script (at least i've never seen rpc use together with a web frontent).

I'm in no way rejecting this (`don't merge` is due to this not beeing ready!) - if finalized it can still add value (as we don't have an alternative to telegram) - but i think investing time in a more versitile protocol (speak rest) could be time better spent.

one point to note (would be the same for a eventual rest implementation: 
can the dependency be optional (similar to plotly) - so if not installed - and this rpc module is disabled - it doesn't cause an issue?

File: freqtrade/rpc/local_rpc_server.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        thread = threading.Thread(target=self.run, args=(freqtrade,)) # extra comma as ref ! Tuple
+        thread.daemon = True                            # Daemonize thread
+        thread.start()                                  # Start the execution
+
+    def run(self, freqtrade):
+        """ Method that runs forever """
+        self._config = freqtrade.config
+
+        # TODO add IP address / port to bind to in config.json and use in below.
+        while True:
```
Comment (xmatthias) [inline freqtrade/rpc/local_rpc_server.py]: is there a need to bind to the server in a loop?

not familiar with zerorpc - but won't `s.run()` remain blocked (until crashed / terminated, anyway)?

how will you exit gracefully from the loop upon bot termination?

Comment (gcarq) [reply freqtrade/rpc/local_rpc_server.py]: After #877 is merged all RPC classes must implement `def cleanup()`, where it should be terminated.

**Comment by gcarq** [comment]
I would also go with REST for the same reasons as mentioned by @xmatthias. This could be done in a few lines with [Tornado](http://www.tornadoweb.org/en/stable/) or [flask](http://flask.pocoo.org/) for example. I'm also not rejecting it if there is the need for it and willingness to maintain.

**Comment by creslinux** [comment]
Good Morning!  TLDR - Rest it is(will be),  While loop not needed (tested)

Thanks for the feedback, 
- on the main comment why-rpc, simply as there was RPC in situ and I thought it would be easier to start from.  Now I'm closer the code in this spot I can see  o use another client api is not huge hurdle.  
I'll take a look at tornado rest

The while loop was code id not realised i could drop - I've tested it this morning and works without which is good to know! Before using a thread in this PR I was playing with greenlets, polling a few other options to wrap zerorpc, that were not working for me, as they blocked the main bot while waiting  - the while was from that

Will look to cleanup in the rest PR

**Pull request closed without merging**